docs/data-sources/firewall.md

```diff
@@ -0,0 +1,84 @@
+---
+page_title: "DigitalOcean: digitalocean_firewall"
+---
+
+# digitalocean_firewall
+
+Get information on a DigitalOcean Firewall.
+
+## Example Usage
+
+Get the firewall:
+
+```hcl
+data "digitalocean_firewall" "example" {
+    firewall_id = "1df48973-6eef-4214-854f-fa7726e7e583"
+}
+
+output "example_firewall_name" {
+    value = data.digitalocean_firewall.example.name
+}
+```
+
+## Argument Reference
+
+* `firewall_id` - (Required) The ID of the firewall to retrieve information
+  about.
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `id` - A unique ID that can be used to identify and reference a Firewall.
+* `status` - A status string indicating the current state of the Firewall.
+  This can be "waiting", "succeeded", or "failed".
+* `created_at` - A time value given in ISO8601 combined date and time format
+  that represents when the Firewall was created.
+* `pending_changes` - A set of object containing the fields, `droplet_id`,
+  `removing`, and `status`.  It is provided to detail exactly which Droplets
+  are having their security policies updated.  When empty, all changes
+  have been successfully applied.
+* `name` - The name of the Firewall.
+* `droplet_ids` - The list of the IDs of the Droplets assigned to
+  the Firewall.
+* `tags` - The names of the Tags assigned to the Firewall.
+* `inbound_rules` - The inbound access rule block for the Firewall.
+* `outbound_rules` - The outbound access rule block for the Firewall.
+
+`inbound_rule` supports the following:
+
+* `protocol` - (Required) The type of traffic to be allowed.
```

Comment:
```suggestion
* `protocol` - The type of traffic to be allowed.
```

Probably makes sense to remove the optional/required bits for these since it is all read only.

Reply:
Got it